### PR TITLE
added [Obsolete] to legacy waiter/listener API.

### PR DESCRIPTION
**Pull request by heejaechang**
we should remove these once partner moved to new API.

### Customer scenario

This is not user facing change. this is for our legacy testing API that partner team still depends on.

### Bugs this fixes

https://github.com/dotnet/roslyn/issues/24574

### Workarounds, if any

N/A

### Risk

[Obsolete] is added. so it will generate warning if partner team still depends on legacy API.

### Performance impact

N/A

### Is this a regression from a previous update?

No

### Root cause analysis

We are moving some legacy test API to new API. new API and shim is already in dev15.7stg, this is follow up change which add [Obsolete] to shim so that we can get rid of those.

### How was the bug found?

N/A

### CODE REVIEW

**Comment by heejaechang** [comment]
can you take a look? @dotnet/roslyn-analysis @dotnet/roslyn-ide @jinujoseph 

this is follow up PR for https://github.com/dotnet/roslyn/pull/24512

File: src/Features/Core/Portable/Shared/TestHooks/Legacy/FeatureAttribute.cs [hunk 1]
```diff
@@ -11,11 +11,13 @@ internal partial class FeatureAttribute : Attribute
     {
         private readonly string _featureName;
 
+        [Obsolete("use IAsynchronousOperationListenerProvider")]
```
Comment (sharwell) [inline src/Features/Core/Portable/Shared/TestHooks/Legacy/FeatureAttribute.cs]: ❔ Move to just having it on the type?

File: src/Features/Core/Portable/Shared/TestHooks/Legacy/AggregateAsynchronousOperationListener.cs [hunk 2]
```diff
@@ -8,6 +8,7 @@
 
 namespace Microsoft.CodeAnalysis.Shared.TestHooks
 {
+    [Obsolete("use IAsynchronousOperationListenerProvider")]
```
Comment (sharwell) [inline src/Features/Core/Portable/Shared/TestHooks/Legacy/AggregateAsynchronousOperationListener.cs]: :memo: This is the type of situation where dotnet/csharplang#384 would provide better assurances regarding the usefulness of the message.

File: src/Features/Core/Portable/Shared/TestHooks/Legacy/FeatureAttribute.cs (hunk 1 above)
Comment (heejaechang) [reply src/Features/Core/Portable/Shared/TestHooks/Legacy/FeatureAttribute.cs]: type is not obsolete. just these methods.

**Comment by heejaechang** [comment]
@Pilchie can I get approval for 15.7?

**Comment by heejaechang** [comment]
tagging @minestarks

File: src/Features/Core/Portable/Shared/TestHooks/Legacy/AggregateAsynchronousOperationListener.cs (hunk 2 above)
Comment (Pilchie) [reply src/Features/Core/Portable/Shared/TestHooks/Legacy/AggregateAsynchronousOperationListener.cs]: Consider using the overload to make it an error.

Comment (heejaechang) [reply src/Features/Core/Portable/Shared/TestHooks/Legacy/AggregateAsynchronousOperationListener.cs]: aren't that be too intrusive? @minestarks

**Comment by heejaechang** [comment]
@Pilchie 
this should binary compatible with inserted typescript if that is what you are asking. I want it to show up as warning next time, typescript, F# build theirs so that they can move to new API.

File: src/Features/Core/Portable/Shared/TestHooks/Legacy/AggregateAsynchronousOperationListener.cs (hunk 2 above)
Comment (heejaechang) [reply src/Features/Core/Portable/Shared/TestHooks/Legacy/AggregateAsynchronousOperationListener.cs]: @minestarks okay checking in?

**Pull request merged**